src/condor_utils/condor_event.cpp

```diff
@@ -331,6 +380,111 @@ const char* ULogEvent::eventName(void) const
+	if (p[0] == '0' && p[1] && p[2] && p[3] == ' ') { p +=3 ; }
+
+	if (p[0] != ' ' || p[1] != '(') return nullptr;
+	p += 2;
+	char * endp = nullptr;
+	cluster = strtol(p, &endp, 10);
+	if (*endp != '.') return nullptr;
+	p = endp+1;
+	proc = strtol(p, &endp, 10);
+	if (*endp != '.') return nullptr;
+	p = endp+1;
+	subproc = strtol(p, &endp, 10);
+	if (*endp != ')') return nullptr;
+	p = endp+1;
+
+	// space before date
+	if (p[0] != ' ') return nullptr;
+	++p;
+
+	// scanning for the next space should either end up pointing to the space after the MM/DD
+	// or to the space after the iso8601 date and time.
+	const char * datend = strchr(p, ' ');
+	if ( ! datend) return nullptr;
+
+	// if we read a / in position 2 of the date, then this must be
+	// a date of the form MM/DD, which is not iso8601. we use the old
+	// (broken) parsing algorithm to parse it, but we can use the iso time parsing code
+	//
+	if (isdigit(p[0]) && isdigit(p[1]) && p[2] == '/') {
+		// here we expect a date of the form MM/DD HH:MM:SS with a space between time and date
+		// datend therefore should point to the space between the date and the time
+		if (datend != p+5) return nullptr;
+		++datend;
+
+		// we can parse the time part as iso8601
+		// this will set -1 into all fields of dt that are not set by time parsing code
+		iso8601_to_time(datend, &dt, &event_usec, &is_utc);
+
+		// we parse the date part as two int fields separated by '/'
+		dt.tm_mon = atoi(p);
+		if (dt.tm_mon <= 0) { // this detects completely garbage dates because atoi returns 0 if there are no numbers to parse
+			return nullptr;
+		}
+		dt.tm_mon -= 1; // recall that tm_mon+1 was written to log
+		dt.tm_mday = atoi(p+3);
+
+		// now scan for the space after the time, so we can return that
+		datend = strchr(datend, ' ');
+	} else if (datend == p+10) {
+		// here we expect the date and time to be in iso8601 format
+		// but the T between date and time is missing.  this is the usual case
+		char datebuf[10 + 1 + 21 + 3]; // longest date is YYYY-MM-DD  = 4+3+3 = 10, longest time is HH:MM:SS.uuuuuu+hh:mm = 7*3 = 21
+		strncpy(datebuf, p, sizeof(datebuf)-1);
+		datebuf[sizeof(datebuf)-1] = 0;
+		// date must be in the form of YYYY-MM-DD and Time must begin at position 11
+		// so we can turn this into a full iso8601 datetime by stuffing a T inbetween
+		datebuf[10] = 'T';
+		// this will set -1 into all fields of dt that are not set by date/time parsing code
+		iso8601_to_time(datebuf, &dt, &event_usec, &is_utc);
+		// now scan for the space after the time, so we can return that
+		datend = strchr(datend+1, ' ');
+	} else {
+		// here we expect the date and time to be in iso8601 format
+		// this will set -1 into all fields of dt that are not set by date/time parsing code
+		iso8601_to_time(p, &dt, &event_usec, &is_utc);
+	}
+
+	// check for a bogus date stamp
+	if (dt.tm_mon < 0 || dt.tm_mon > 11 || dt.tm_mday < 0 || dt.tm_mday > 32 || dt.tm_hour < 0 || dt.tm_hour > 24) {
```

Comment:
Is 24 valid for dt.tm_hour here?

Reply:
I think so.  this line is from the old code.

Reply:
in fact, from here down this is from the old code except for the last 2 lines.